### ISSUE TITLE
Starting and stopping MySql, mysql.sock?

### ISSUE THREAD

**Issue by frumbert** [pbougie/mac-dev-env#8]
MySql is falling over all the time, and I don't know why.

I haven't edited `my.cnf` at all since installing via your instructions.

The MySQL preference panel says my server is stopped, but I'm happily using it with Sequel Pro. It posits the warning `The /usr/local/mysql/data directory is not owned by the 'mysql' or '_mysql' user`, but my dataroot is actually /usr/local/var/mysql/ (which is owned by tim:staff; that is, me). I guess the pref panel doesn't work with this install method.

The console is showing this message every 10 seconds:

    8/07/2015 8:03:44.416 pm com.apple.xpc.launchd[1]: (com.mysql.mysqld) Service only ran for 0 seconds. Pushing respawn out by 10 seconds.
    8/07/2015 8:03:52.365 pm com.apple.xpc.launchd[1]: (org.apache.httpd) Service only ran for 0 seconds. Pushing respawn out by 10 seconds.

which is strange as the mysqld process is actually running. However, in my activity monitor I can see the mysqld process which is owned by tim (me). Every now and then, another one pops up which is owned by root, then vanishes before I can examine it. May be related, but I can't tell.

    Tims-MacBook-Pro:mysql tim$ ps -e | grep mysql
    14110 ??         0:00.01 /bin/sh /usr/local/mysql/bin/mysqld_safe --user=mysql --datadir=/usr/local/var/mysql --log-error=/usr/local/var/log/mysql.log
    14218 ??         0:00.41 /usr/local/mysql-5.6.25/bin/mysqld --basedir=/usr/local/mysql-5.6.25 --datadir=/usr/local/var/mysql --plugin-dir=/usr/local/mysql-5.6.25/lib/plugin --log-error=/usr/local/var/log/mysql.log --pid-file=/usr/local/var/mysql/Tims-MacBook-Pro.local.pid
    14310 ttys001    0:00.00 grep mysql

I checked the launchagents to see if there wasn't something sneaky in there like a second instance - no.

    Tims-MacBook-Pro:mysql tim$ ls -la ~/Library/LaunchAgents/
    total 64
    drwxr-xr-x  10 tim  staff   340  5 Jul 20:17 .
    drwx------@ 57 tim  staff  1938  7 Jul 13:54 ..
    -rw-r--r--   1 tim  staff   697 23 Jun 23:23 com.adobe.AAM.Updater-1.0.plist
    -rw-r--r--@  1 tim  staff   800 23 Jun 22:54 com.google.keystone.agent.plist
    -rw-r--r--   1 tim  staff   759  5 Jul 20:17 com.mysql.mysqld.plist
    -rw-r--r--   1 tim  staff   807  4 Jul 21:17 com.valvesoftware.steamclean.plist
    -rw-r--r--   1 tim  staff   662 29 Jun 12:57 io.redis.redis-server.plist
    -rw-r--r--   1 tim  staff   708 29 Jun 12:54 org.memcached.memcached.plist
    -rw-r--r--   1 tim  staff   694 29 Jun 12:12 org.mongodb.mongod.plist
    -rw-r--r--   1 tim  staff   677 24 Jun 00:17 org.virtualbox.vboxwebsrv.plist
    Tims-MacBook-Pro:mysql tim$ sudo ls -la /Library/LaunchAgents/
    Password:
    total 40
    drwxr-xr-x   7 root  wheel   238  8 Jul 19:44 .
    drwxr-xr-x+ 63 root  wheel  2142 28 Jun 19:33 ..
    -rw-r--r--   1 root  wheel   612 23 Jun 23:27 com.adobe.AAM.Updater-1.0.plist
    -rw-r--r--   1 root  wheel   667 23 Jun 23:02 com.adobe.AdobeCreativeCloud.plist
    lrwxr-xr-x   1 root  wheel   104 24 Jun 00:08 com.oracle.java.Java-Updater.plist -> /Library/Internet Plug-Ins/JavaAppletPlugin.plugin/Contents/Resources/com.oracle.java.Java-Updater.plist
    -rw-r--r--   1 root  wheel   668  3 Mar 00:11 com.teamviewer.teamviewer.plist
    -rw-r--r--   1 root  wheel   779  3 Mar 00:11 com.teamviewer.teamviewer_desktop.plist

just the usual suspects. I'd like to stop mysql and start it again. But

    Tims-MacBook-Pro:support-files tim$ mysqladmin --user=root --password shutdown
    Enter password: 
    mysqladmin: connect to server at 'localhost' failed
    error: 'Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)'
    Check that mysqld is running and that the socket: '/tmp/mysql.sock' exists!

I can't shut it down, becuase there's no socket file. This cropped up earlier and I ignored it since my apps were trying to connect to `localhost` and wouldn't until I changed them over to `127.0.0.1` .. because `localhost` assumes a socket, which isn't there. So your last step to test the server fails:

    Tims-MacBook-Pro:support-files tim$ mysql --user=root --password
    Enter password: 
    ERROR 2002 (HY000): Can't connect to local MySQL server through socket '/tmp/mysql.sock' (2)

The only sock I seem to have is in /private

    Tims-MacBook-Pro:support-files tim$ locate mysql.sock
    /private/var/mysql/mysql.sock

Pointers appreciated here. My expertise is more in the sql than the server ...

**Comment by pbougie** [comment]
I’m not seeing any glaring problems. To shut down MySQL, you need to remove it from Launchd:
launchctl unload ~/Library/LaunchAgents/com.mysql.mysqld.plist
My suggestion is to delete the installation once you’re sure it’s completely stopped, including the program files, data files (backup if required) and the socket file. Keep an eye on the process using Activity Monitor. When you re-install, also keep an eye on Console log messages for any problems.

You also seem to be having a problem with Apache. When Launchd can’t run a service, it means something is wrong with the installation. Unload the service from Launchd, re-install, run it manually and check if it works.

Good luck,
Patrick

**Comment by frumbert** [comment]
After much swearing I discovered that the internal apache was trying to fire up. So as soon as I unloaded and stopped httpd, it would magically appear again - it was `/System/Library/LaunchDaemons/org.apache.apache2.plist` - so that's been unloaded and now apache isn't complaining so much. I deleted and recreated the plists, and discovered that I'd put mysql into launchdamons as well. doh! so now both apache and mysql are launching *once*. I symlinked `/tmp/mysql.sock` (which didn't exist) to `/usr/local/mysql/mysql.sock` (which also didn't exist) and suddenly sockets are working. funny that. grrr
